src/runtime-rs/crates/runtimes/src/tracer.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::cmp::min;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use anyhow::Result;
+use lazy_static::lazy_static;
+use opentelemetry::global;
+use opentelemetry::runtime::Tokio;
+use tracing::{span, subscriber::NoSubscriber, Span, Subscriber};
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::Registry;
+
+lazy_static! {
+    /// The static KATATRACER is the variable to provide public functions to outside modules
+    static ref KATATRACER: Arc<Mutex<KataTracer>> = Arc::new(Mutex::new(KataTracer::new()));
+
+    /// The ROOTSPAN is a phantom span that is running by calling [`trace_enter_root()`] at the background
+    /// once the configuration is read and config.runtime.enable_tracing is enabled
+    /// The ROOTSPAN exits by calling [`trace_exit_root()`] on shutdown request sent from containerd
+    ///
+    /// NOTE:
+    ///     This allows other threads which are not directly running under some spans to be tracked easily
+    ///     within the entire sandbox's lifetime.
+    ///    To do this, you just need to add attribute #[instrment(parent=&(*ROOTSPAN))]
+    pub static ref ROOTSPAN: Span = span!(tracing::Level::TRACE, "root-span");
+}
+
+/// The tracer wrapper for kata-containers, this contains the global static variable
+/// the tracing utilities might need
+/// The fields and member methods should ALWAYS be PRIVATE and be exposed in a safe
+/// way to other modules
+unsafe impl Send for KataTracer {}
+unsafe impl Sync for KataTracer {}
+struct KataTracer {
+    subscriber: Arc<dyn Subscriber + Send + Sync>,
+    enabled: bool,
+}
+
+//todo: remove hyper's tracing
```

Comment:
?

Reply:
Since some crates like `hyper` are also emitting traces to jaeger's collector, we may offer a way to filter out any traces from other crates.

But I have not found an elegant way of doing this, so currently, other traces will show separately w.r.t ROOTSPAN(which we really care about). Luckily, the ROOTSPAN is not affected and the spans under it are clean.